lib/ppx/ppx_js.cppo.ml

```diff
@@ -23,43 +23,35 @@ let exp_to_string = function
 let lid ?(loc= !default_loc) str =
   Location.mkloc (Longident.parse str) loc
 
-let mk_loc ?(loc= !default_loc) x = Location.mkloc x loc
-
-let mk_id ?loc str =
-  let exp = Exp.ident ?loc (lid ?loc str)
-  and pat = Pat.var ?loc (mk_loc ?loc str) in
-  (exp, pat)
-
-
 (** arg1 -> arg2 -> ... -> ret *)
 let arrows args ret =
   List.fold_right (fun (l, ty) fun_ -> Typ.arrow l ty fun_)
     args
     ret
 
-(** fun arg1 arg2 ... -> ret *)
-let funs args ret =
-  List.fold_right (fun pat next_fun_ -> lam pat next_fun_)
-    args
-    ret
-
-let rnd = Random.State.make [|0x313511d4|]
-let random_var () =
-  Format.sprintf "jsoo_%08Lx" (Random.State.int64 rnd 0x100000000L)
-let random_tvar () =
-  Format.sprintf "jsoo_%08Lx" (Random.State.int64 rnd 0x100000000L)
+let targs_arrows targs =
+  List.map (fun (l,args,res) -> l, arrows args res) targs
 
 let inside_Js = lazy
   (try
      Filename.basename (Filename.chop_extension !Location.input_name) = "js"
    with Invalid_argument _ -> false)
 
+let merlin_noloc = { txt = "merlin.loc"; loc = Location.none }, PStr []
+
+(* let to_js_of_ocaml = lazy (Some "Js_of_ocaml") *)
+let to_js_of_ocaml = lazy None
```

Comment:
It's indeed used inside Janestreet. I'll remove this diff as it does not belong in the PR.